compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt

```diff
@@ -381,10 +390,55 @@ class ExpressionCodegen(
         get() = origin != IrDeclarationOrigin.IR_TEMPORARY_VARIABLE &&
                 origin != IrDeclarationOrigin.FOR_LOOP_ITERATOR
 
+    private fun AbstractInsnNode.computeOffset(): Int {
```

Comment:
> Let me know if you think of a better way of doing it!
@madsager  I will look, looks like it's also worth to mention @pyos (he always have some fresh ideas)

Reply:
Yeah, we were just talking and since this should be always nested gaps inside of other gaps, maybe we can avoid generating the nested gap in the first place. However, the trivial change to do that doesn't work in the "return from nested finally with a non-local return case" (tryFinally9 test case). I'll look more next week. :)

Reply:
I was just holding it wrong. Updated to not generate nested gaps in the first place which resolves this issue.

Reply:
> Apparently, there is poor test coverage of nested try-finally blocks

Yes, sadly, some parts have bad coverage. 

Thank you for adding new tests with inline in case of non-local return in catch/finally (as was mentioned in my original asking above). 
Could you also add similar test without inline functions for similar cases in catch/finally? (Sorry for misleading in original answer, it was updated in short time after posting)

Reply:
Thanks @max-kammerer. I added some more tests in the third commit. Are these what you were looking for?

These tests show that exception variables are not properly tracked and therefore they bleed into the finally blocks. Also, there are some interesting line stepping going on. I have added comments in the files. I would like to not fix these in this change. Once we are happy with this change, I'll file YouTrack issues for the TODOs in the test files and add them to the list of things to work on.

Reply:
@madsager Thank you!